### ISSUE TITLE
Potential bug in automated weight assignment for non-weighted empirical distributions

### ISSUE THREAD

**Issue by gbaydin** [DiffSharp/DiffSharp#394]
The code here

https://github.com/DiffSharp/DiffSharp/blob/5a80ff16aab64021dbab8280dee7b95b311940e6/src/DiffSharp.Core/Distributions.fs#L259

might have a bug. The code seems to be initializing a categorical distribution with logits (unnormalized logprobabilities) but the code is initializing this to the counts of unique values. At first sight, this looks like it should be taking the log of counts instead. Needs to be checked and represented in the unit tests.

**Comment by gbaydin** [comment]
Fixed in #416
